docs/drupal_xdebug.md

```diff
@@ -0,0 +1,48 @@
+# Drupal Twig Debugging
+
+### Locating Template files
+
+1. Copy the `docroot/sites/default/example.services.local.yml` file to `docroot/sites/default/services.local.yml`.
+2. Clear the cache by running `ddev drush cr`.
+3. Inspect the rendered page in a browser.
+4. You should see HTML comments indicating what templates are being used throughout the source code. See [Drupal.org: Locating Template Files with Debugging](https://www.drupal.org/node/2358785)
+
+### Quickest Debug Method: Twig Tweak Drupal Dump
+
+* Add `{{ drupal_dump() }}`, `{{ drupal_dump(variable-name) }}` or `{{ dd() }}` or `{{ dd(variable-name) }}` to a template file and refresh (May require a cache clear). See [Drupal.org: Twig Tweak Cheat sheet (Drupal Dump)](https://www.drupal.org/docs/contributed-modules/twig-tweak-2x/cheat-sheet#s-drupal-dump)
+* You can add this to multiple times to the same or different template files.
+
+**Note: Remember to remove all calls to `drupal_dump()` or `dd()` from templates before committing.**
+
+### For Mass.gov using Xdebug and PhpStorm
+
+*Initial setup:*
+1. Set up Xdebug in PhpStorm: https://www.jetbrains.com/help/phpstorm/configuring-xdebug.html
+2. Enable Xdebug in ddev by running `ddev xdebug on`.  See [PHP Step Debugging](https://ddev.readthedocs.io/en/stable/users/step-debugging/) for ddev specific setup instructions and how to configure your IDE
+3. Download and enable the [Xdebug helper](https://www.jetbrains.com/help/phpstorm/browser-debugging-extensions.html) browser extension.
+
+*For Twig debugging, follow these additional steps:*
+1. Copy the `docroot/sites/default/example.services.local.yml` file to `docroot/sites/default/services.local.yml`.
+3. Clear the cache by running `ddev drush cr`.
+4. In PhpStorm, under Preferences > PHP > Debug > Templates > Twig Debug, set the Cache path (e.g. `/Users/username/Projects/openmass/docroot/sites/default/files/php/twig`).
```

Comment:
I want to see if I can get the xdebug part working. And I think I'm having some issues with the path configuration here. Do we have to specify the path of the file that is being tested? I'm not sure if I understand this example. 

For example, I'm adding xdebug/devel breakpoints in this file, /Users/minghuasun/Documents/Github/openmass/docroot/themes/custom/mass_theme/templates/layout/html.html.twig. Is this the path I should add to cache path? So each time I debug a file I need to change this path in the preference? 

<img width="454" alt="Screen Shot 2021-12-08 at 6 04 17 PM" src="https://user-images.githubusercontent.com/5789411/145305807-b0449fa5-6d38-49b2-a38b-c42e47bb4db7.png">
<img width="424" alt="Screen Shot 2021-12-08 at 6 06 14 PM" src="https://user-images.githubusercontent.com/5789411/145305808-8e677552-8394-4109-8f03-1b8027a626be.png">